pkg/action/failback.go

```diff
@@ -211,4 +209,270 @@ func (ac *ActionController) performDeactivateFailBackStage(action *storkv1.Actio
 			return
 		}
 	}
+	log.ActionLog(action).Infof("Going to update the action status successful for stage %s", storkv1.ActionStageScaleDownDestination)
+	action.Status.Status = storkv1.ActionStatusSuccessful
+	action.Status.Reason = ""
+	ac.updateAction(action)
+}
+
+func (ac *ActionController) performLastMileMigration(action *storkv1.Action) {
+	if action.Status.Status == storkv1.ActionStatusSuccessful {
+		action.Status.Stage = storkv1.ActionStageScaleUpSource
+		action.Status.Status = storkv1.ActionStatusInitial
+		action.Status.FinishTimestamp = metav1.Now()
+		ac.updateAction(action)
+		return
+	} else if action.Status.Status == storkv1.ActionStatusFailed {
+		action.Status.Stage = storkv1.ActionStageScaleUpDestination
+		action.Status.Status = storkv1.ActionStatusInitial
+		action.Status.FinishTimestamp = metav1.Now()
+		ac.updateAction(action)
+		return
+	}
+
+	migrationSchedule, err := storkops.Instance().GetMigrationSchedule(action.Spec.ActionParameter.FailbackParameter.MigrationScheduleReference, action.Namespace)
+	if err != nil {
+		msg := fmt.Sprintf("error fetching migrationschedule %s/%s for failback", action.Namespace, action.Spec.ActionParameter.FailbackParameter.MigrationScheduleReference)
+		log.ActionLog(action).Errorf(msg)
+		action.Status.Status = storkv1.ActionStatusFailed
+		action.Status.Reason = msg
+		ac.recorder.Event(action,
+			v1.EventTypeWarning,
+			string(storkv1.ActionStatusFailed),
+			msg)
+		ac.updateAction(action)
+		return
+	}
+
+	migrationName := getLastMileMigrationName(migrationSchedule.Name, string(storkv1.ActionTypeFailback), utils.GetShortUID(string(action.UID)))
+	migrationObject, err := storkops.Instance().GetMigration(migrationName, migrationSchedule.Namespace)
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			// If last mile migration CR does not exist, create one
+			migration := getLastMileMigrationSpec(migrationSchedule, string(storkv1.ActionTypeFailback), utils.GetShortUID(string(action.UID)))
+			_, err := storkops.Instance().CreateMigration(migration)
+			if err != nil {
+				msg := fmt.Sprintf("Creating last mile migration from migrationschedule %s failed: %v", migrationSchedule.GetName(), err)
+				ac.recorder.Event(action,
+					v1.EventTypeWarning,
+					string(storkv1.ActionStatusFailed),
+					msg)
+				ac.updateAction(action)
+				return
+			}
+			action.Status.Status = storkv1.ActionStatusInProgress
+			ac.updateAction(action)
+			return
+		} else {
+			msg := fmt.Sprintf("Failed to get last mile migration object %s: %v", migrationName, err)
+			log.ActionLog(action).Errorf(msg)
+			ac.recorder.Event(action,
+				v1.EventTypeWarning,
+				string(storkv1.ActionStatusFailed),
+				msg)
+			ac.updateAction(action)
+			return
+		}
+	}
+
+	if migrationObject.Status.Stage == storkv1.MigrationStageFinal {
+		if migrationObject.Status.Status == storkv1.MigrationStatusSuccessful || migrationObject.Status.Status == storkv1.MigrationStatusPartialSuccess {
```

Comment:
We will update to ActionStatusSuccessful even in case of MigrationStatusPartialSuccess?